### DISCUSSION TITLE
Is altitude taken into account ?

### DISCUSSION THREAD

**Discussion by gabriel-mercier** [jpb10/SolarCalculator#27] [Q&A]
Hello,

Thank you for sharing this library! I just have a quick question: does the calcHorizontalCoordinates() function take altitude into account? For instance, for the same latitude and longitude but at different altitudes, let's say 0m and 4000m, the elevation for the sun will be different. Is is implemented or does it need to be manually calculated by us ?

Thanks!

**Comment by jpb10** [comment]
Hello,

It's not necessary... at 4000 meters, the correction in elevation would be < 0.0000016 degrees. (That said, it does make a difference for sunrise and sunsets because it changes the level of the horizon!) 

The calculation is called correction for parallax. The observer's altitude can be ignored since it is dwarfed by the Earth's radius, itself dwarfed by the Sun's distance. 

The correction for parallax in azimuth is very small because the Earth is nearly a sphere. The correction for parallax in altitude $p$ can be approximated  (Meeus, 1998, ch. 40):

$p = \rho \ \pi \cos{(h)} \approx 8^".794 \cos{(h)}$

where $\rho$ is the observer's distance from the Earth's center (equatorial radius $= 1$); $\pi$ the equatorial horizontal parallax; and $h$ the body's altitude (i.e. elevation). The correction $p$ is always < 0.0025 degrees and can be ignored for our purposes.

**Reply by gabriel-mercier** [reply]
Thank you very much !
